kbase-extension/static/kbase/js/widgets/narrative_core/upload/stagingAreaViewer.js

```diff
@@ -439,294 +495,399 @@ define([
-                        } else {
-                            importDropdown.select2({
-                                placeholder: 'Select a type',
-                                containerCssClass: `${cssBaseClass}-body__import-dropdown`,
-                            });
-                        }
-                    }
+                //make checkbox for that row enabled
+                //also set the data type so that we have the reference later when importing
+                $checkboxElem
+                    .prop('disabled', false)
+                    .attr('aria-label', 'Select to import file checkbox')
+                    .attr('data-type', dataType);
 
-                    //set the behavior on the import dropdown when a user selects a type
-                    importDropdown.on('select2:select', (e) => {
-                        const dataType = e.currentTarget.value;
+                //make sure select all checkbox is enabled
+                $('#staging_table_select_all')
+                    .prop('disabled', false)
+                    .attr('aria-label', 'Select to import all files checkbox');
 
-                        //store the type we selected along with file data so we can persist on a view update
-                        rowFileData.dataType = dataType;
-                        stagingAreaViewer.selectedFileTypes[rowFileName] = rowFileData;
+                if (check) {
+                    $checkboxElem.prop('checked', true).attr('aria-checked', true).change();
+                    stagingAreaViewer.enableImportButton();
+                }
+            }
+
+            const storedFileData = this.selectedFileTypes[data.path];
+            // where we have data type set, render those dropdowns correctly
+            if (storedFileData) {
+                // tell select2 which option to set
+                importDropdown
+                    .select2({
+                        containerCssClass: `${cssBaseClass}-body__import-dropdown`,
+                    })
+                    .val(storedFileData.dataType)
+                    .trigger('change');
+                //enable the checkboxes
+                enableCheckboxes(storedFileData.dataType);
+            }
 
-                        enableCheckboxes(dataType, true);
+            //otherwise we set the dropdowns with a placeholder
+            else {
+                // And if we have a suggested type, select it and enable the checkboxes
+                if (suggestedType) {
+                    importDropdown
+                        .select2({
+                            containerCssClass: `${cssBaseClass}-body__import-dropdown ${cssBaseClass}-body__import-type-selected`,
+                            placeholder: 'make the empty option disappear',
+                        })
+                        .val(suggestedType.id)
+                        .trigger('change');
+                    enableCheckboxes(suggestedType.id);
```

Comment:
not too major, but you could use an `else if`, which would allow you to decrease the indent level to improve readability a little.